AppHandling/Publish-NavContainerApp.ps1

```diff
@@ -328,7 +328,21 @@ try {
             
                     if ($publishIt) {
                         Write-Host "Publishing $appFile"
-                        Publish-NavApp -ServerInstance $ServerInstance -Path $appFile -SkipVerification:$SkipVerification @publishArgs
+                        $publishArgs += @{"serverInstance" = $serverInstance; "Path" = $appFile; "SkipVerification" = $skipVerification.IsPresent}
+                        if ($PSVersionTable.PSVersion.Major -lt 7 -and $version.Major -ge 24) {
+                            # GIANT HACK AHEAD...
+                            # To cope with this issue https://github.com/microsoft/navcontainerhelper/issues/3591 where the PS5 bridge to PS7 hangs when calling Publish-NavApp directly
+                            # and due to https://github.com/microsoft/navcontainerhelper/issues/3575 and https://github.com/PowerShell/PowerShell/issues/23982 we cannot use -usepwsh:$true
+                            $command = '$serviceTierFolder = (Get-Item ''C:\Program Files\Microsoft Dynamics NAV\*\Service'').FullName;Import-Module (Join-Path $serviceTierFolder ''Admin\Microsoft.Dynamics.Nav.Management.psm1'');Import-Module (Join-Path $serviceTierFolder ''Admin\Microsoft.BusinessCentral.Management.psd1'');Import-Module (Join-Path $serviceTierFolder ''Admin\Microsoft.BusinessCentral.Apps.Management.dll'');'

```

Comment:
Would it be possible to move this code out to its own .ps1 file (with parameters), and invoke that one (via pwsh)? It would make it a bit more readable.

Reply:
I could - and I will if I need to do this with other commands - but until then I will keep it here, clearly showing that the hack is only used here.